src/gsi/general_specmod.f90

```diff
@@ -306,6 +307,45 @@ subroutine general_init_spec_vars(sp,jcap,jcap_test,nlat_a,nlon_a,eqspace)
     return
   end subroutine general_init_spec_vars
 
+  subroutine general_spec_multwgt(sp,spcwrk,spcwgt)
+!   2017-03-30  J. Kay, X. Wang - add general_spec_multwgt for scale-dependent
+!   weighting of mixed resolution ensemble,
+!                                 POC: xuguang.wang@ou.edu
+!
+    implicit none
+    type(spec_vars),intent(in) :: sp
+    real(r_kind),dimension(sp%nc),intent(inout) :: spcwrk
+    real(r_kind),dimension(0:sp%jcap),intent(in) :: spcwgt
+
+    integer(i_kind) ii1,l,m,jmax,ks,n
+
+!    ii1=0
+!    do l=0,sp%jcap
+!       do m=0,sp%jcap-l
+!          ii1=ii1+2
+!          spcwrk(ii1-1)=spcwrk(ii1-1)*spcwgt(l)
+!          spcwrk(ii1)=spcwrk(ii1)*spcwgt(l)
+!       end do
+!    end do
+!! Code borrowed from spvar in splib
+    JMAX=sp%jcap
+
+    L=0
+    DO N=0,JMAX
+       KS=2*N
+       spcwrk(KS+1)=spcwrk(KS+1)*spcwgt(N)
+    ENDDO
+    DO N=0,JMAX
+       DO L=MAX(1,N-JMAX),MIN(N,JMAX)
+          KS=L*(2*JMAX+(-1)*(L-1))+2*N
+          spcwrk(KS+1) = spcwrk(KS+1)*spcwgt(N)
+          spcwrk(KS+2) = spcwrk(KS+2)*spcwgt(N)
+       ENDDO
+    ENDDO
+
+    return
+  end subroutine general_spec_multwgt
+
```

Comment:
better to use  lower cases for "do ... enddo" instead of "DO ... ENDDO".
Try to use lower case for consistent with rest of the code.

Reply:
Changed.